src/Categories/Multi/Category/Cartesian/Representable.agda

```diff
@@ -0,0 +1,174 @@
+
+  open Category C
+  open Equiv
+  open HomReasoning
+  open Cartesian cart
+  open ×P C
+  open Prods cart
+
+  infix 4 _⇒Π_ _≈Π_
+
+  -- A map of type `X ⇒ prod Δ` is equivalently
+  -- a map of type `X ⇒ Y` for each Y ∈ Δ.
+  -- `X ⇒Π Δ` is exactly the latter.
+
+  _⇒Π_ : Obj → List Obj → Set (o ⊔ ℓ)
+  X ⇒Π Δ = Env (X ⇒_) Δ
+
+  _≈Π_ : ∀ {X Δ} (fs gs : X ⇒Π Δ) → Set (o ⊔ e)
+  _≈Π_ = [ _≈_ ]_≈ᵉ_
+
+  ⟨_⟩Π : ∀ {X Δ} → X ⇒Π Δ → X ⇒ prod Δ
+  ⟨_⟩Π {Δ = []} fs = !
+  ⟨_⟩Π {Δ = Y ∷ Δ} fs = ⟨ fs (here ≡.refl) , ⟨ fs F.∘ there ⟩Π ⟩
+
+  ⟨_⟩Π-resp-≈ : ∀ {X Δ} {fs gs : X ⇒Π Δ} → fs ≈Π gs → ⟨ fs ⟩Π ≈ ⟨ gs ⟩Π
+  ⟨_⟩Π-resp-≈ {Δ = []} ps = refl
+  ⟨_⟩Π-resp-≈ {Δ = Y ∷ Δ} ps =
+    ⟨⟩-cong₂ (ps .get (here ≡.refl)) ⟨ [ ps .get F.∘ there ] ⟩Π-resp-≈
+
+  projectΠ : ∀ {X Y Δ} (i : Y ∈ Δ) (fs : X ⇒Π Δ) → π[ i ] ∘ ⟨ fs ⟩Π ≈ fs i
+  projectΠ (here ≡.refl) fs = project₁
+  projectΠ (there i) fs =
+    trans assoc (trans (∘-resp-≈ refl project₂) (projectΠ i _))
+
+  ⟨π⟩Π : ∀ {Γ} → ⟨ π[_] {_} {Γ} ⟩Π ≈ id
+  ⟨π⟩Π {Γ} = uniqueness* {ys = Γ} λ i → trans (projectΠ i _) (sym identityʳ)
+
+  Rep : CartesianMultiCategory o ℓ e
+  Rep = record
+    { Obj = Obj
+    ; _⇒_ = λ Γ A → prod Γ ⇒ A
+    ; _≈_ = _≈_
+    ; id = π[_]
+    ; _∘_ = λ f σ → f ∘ ⟨ σ ⟩Π
+    ; equiv = equiv
+    ; ∘-resp-≈ = λ ff σσ → ∘-resp-≈ ff ⟨ σσ ⟩Π-resp-≈
+    ; identityˡ = λ {_} {_} {_} {i} → projectΠ i _
+    ; identityʳ = λ {Γ} → trans (∘-resp-≈ʳ (⟨π⟩Π {Γ})) identityʳ
+    ; identity² = λ {_} {_} {i} → projectΠ i _
+    ; assoc = λ {Γ Δ Θ A f σ τ} → begin
+      (f ∘ ⟨ σ ⟩Π) ∘ ⟨ τ ⟩Π  ≈⟨ assoc ⟩
+      f ∘ (⟨ σ ⟩Π ∘ ⟨ τ ⟩Π)
+        ≈⟨ refl⟩∘⟨ (uniqueness* {ys = Θ} λ i → begin
+        π[ i ] ∘ ⟨ σ ⟩Π ∘ ⟨ τ ⟩Π  ≈⟨ sym-assoc ⟩
+        (π[ i ] ∘ ⟨ σ ⟩Π) ∘ ⟨ τ ⟩Π  ≈⟨ projectΠ i _ ⟩∘⟨refl ⟩
+        σ i ∘ ⟨ τ ⟩Π  ≈˘⟨ projectΠ i _ ⟩
+        π[ i ] ∘ ⟨ (λ j → σ j ∘ ⟨ τ ⟩Π) ⟩Π  ∎) ⟩
+      f ∘ ⟨ (λ i → σ i ∘ ⟨ τ ⟩Π) ⟩Π  ∎
+    ; sym-assoc = λ {Γ Δ Θ A f σ τ} → begin
+      f ∘ ⟨ (λ i → σ i ∘ ⟨ τ ⟩Π) ⟩Π
+        ≈⟨ refl⟩∘⟨ (uniqueness* {ys = Θ} λ i → begin
+        π[ i ] ∘ ⟨ (λ j → σ j ∘ ⟨ τ ⟩Π) ⟩Π  ≈⟨ projectΠ i _ ⟩
+        σ i ∘ ⟨ τ ⟩Π  ≈˘⟨ projectΠ i _ ⟩∘⟨refl ⟩
+        (π[ i ] ∘ ⟨ σ ⟩Π) ∘ ⟨ τ ⟩Π  ≈⟨ assoc ⟩
+        π[ i ] ∘ ⟨ σ ⟩Π ∘ ⟨ τ ⟩Π  ∎) ⟩
+      f ∘ (⟨ σ ⟩Π ∘ ⟨ τ ⟩Π)  ≈⟨ sym-assoc ⟩
+      (f ∘ ⟨ σ ⟩Π) ∘ ⟨ τ ⟩Π  ∎
+    }
+
+-- The *free* Cartesian category generated by a Cartesian multicategory is the
+-- category in which objects are lists of objects from the multicategory, and
+-- morphisms are given by _⇒ˢ_ (simultaneous maps, a multimap for each object
+-- in the codomain).
+-- If the multicategory is given by the syntax of a simple type theory, with
+-- `Γ ⇒ A` being the set of *terms* of type `A` in context `Γ`, then the free
+-- Cartesian category is the category of contexts and simultaneous
+-- substitutions.
+
+module _ (CM : CartesianMultiCategory o ℓ e) where
+  open CartesianMultiCategory CM
+  open Equiv
+  open Equivˢ
+
+  Free : ∃ \ C → Cartesian C
+  Free = record
```

Comment:
It would be a lot more readable to define the `Category` instance and then the `Cartesian` instance in either `private` or `where blocks, and then use `_,_` to pair them.  Or maybe use the `CartesianCategory` bundle?